From c7df112e7d92add0865f72905dad050ef3e994d5 Mon Sep 17 00:00:00 2001
From: Russ Hamilton <behamilton@google.com>
Date: Fri, 11 Oct 2024 13:24:10 +0000
Subject: [PATCH] Fix crash in ObliviousHttpRequestHandler

Change-Id: Iec71d84554af8757aa2c032f2565328ac3931f3d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5923384
Reviewed-by: Maks Orlovich <morlovich@chromium.org>
Auto-Submit: Russ Hamilton <behamilton@google.com>
Commit-Queue: Maks Orlovich <morlovich@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1367487}
---

--- a/services/network/oblivious_http_request_handler.cc
+++ b/services/network/oblivious_http_request_handler.cc
@@ -59,6 +59,9 @@
     if (!key_configs.ok()) {
       return std::nullopt;
     }
+    if (key_configs->NumKeys() == 0) {
+      return std::nullopt;
+    }
     quiche::ObliviousHttpHeaderKeyConfig key_config =
         key_configs->PreferredConfig();
     auto ohttp_client = quiche::ObliviousHttpClient::Create(
--- a/services/network/oblivious_http_request_handler_unittest.cc
+++ b/services/network/oblivious_http_request_handler_unittest.cc
@@ -454,6 +454,18 @@
   }
 }
 
+TEST_F(TestObliviousHttpRequestHandler, TestEmptyKeyConfig) {
+  std::unique_ptr<network::ObliviousHttpRequestHandler> handler =
+      CreateHandler();
+  network::mojom::ObliviousHttpRequestPtr request = CreateRequest();
+  request->key_config = "";
+
+  TestOhttpClient client;
+  client.SetExpectedNetError(net::ERR_INVALID_ARGUMENT);
+  handler->StartRequest(std::move(request), client.CreatePendingRemote());
+  client.WaitForCall();
+}
+
 TEST_F(TestObliviousHttpRequestHandler, TestRequestFormat) {
   std::unique_ptr<network::ObliviousHttpRequestHandler> handler =
       CreateHandler();
